docs/contributing.rst

```diff
@@ -2,12 +2,122 @@
 
 .. _contributing:
 
+.. toctree::
+ :maxdepth: 2
+ :numbered:
+
 
 Contributor's Guide
 ===================
 
-Anyone can contribute to and participate in the Uxarray project
-at any levels of project development! We conduct all of our work
-in the open, and all of our work is Open Source Licensed.
+Welcome to the Contributor's Guide for UXarray!
+
+So, welcome to the team! If you are reading this document, we hope that
+you are already or soon-to-be a UXarray contributor! Please keep reading!
+
+1. Overview
+-----------
+
+UXarray is a community-owned, open-development effort that is a result
+of the collaboration between `Project Raijin <https://raijin.ucar.edu/>`_
+and DOE’s `SEATS Project <https://seatstandards.org/>`_ ]. Even though the
+Uxarray team has initiated and been expanding this package, outside
+contributions are welcome and vital for its long-term sustainability.
+Therefore, we invite other community members to become part of this
+collaboration at any level of contribution.
+
+1.1. Why the name "UXarray"?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We have created UXarray by composing `Xarray <https://docs.xarray.dev/en/stable/>`_,
+a Pangeo ecosystem package commonly-used  for structured grids
+recognition, to support reading and recognizing unstructured grid
+model outputs. We picked the name "UXarray" and preferred to
+capitalize the first two letters to emphasize it is Xarray for
+Unstructured grids.
+
+1.2. Many ways to contribute
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are many different ways to contribute to UXarray. Anyone can, for
+example,
+
+- Write or revise user documentation (including this document)
+
+- Implement data analysis operators for unstructured grids from scratch
+  or from their workflows
+
+- Develop example notebooks that demonstrate how a particular function
+  is used
+
+- Answer a support question
+
+- Simply request a feature or report a bug
+
+All of these activities are signicant contributions to the on-going
+development and maintenance of UXarray.
+
+1.3. About this guide
+^^^^^^^^^^^^^^^^^^^^^
+
+The UXarray content is hosted on GitHub, and through this document,
+we aim to ease the community's experience with contributing to this project.
+However, this guide might still be missing case-specific details; please do
+not hesitate to reach out to us to consult any such cases.
+
+Lastly, much of the information in this guide has been co-opted from the
+`GeoCAT <https://geocat.ucar.edu/pages/contributing.html>`_ project and
+`Project Pythia <https://projectpythia.org/contributing.html>`_.
+
+1.4. Project Specific Resources
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Some important UXarray resources are as follows:
+
+- `Uxarray GitHub repository <https://github.com/UXARRAY/uxarray>`_ houses
+  the open-source code base along with some significant documentation such
+  as “Readme”, “Installation”, “Contributing”, and UXarray draft API.
```

Comment:
Should we make these links?

Reply:
Thanks for pointing this out! I also noticed we should provide UXarray documentation as another resource here and link to Installation and Contributing links as well as some others from under that.